RecoParticleFlow/PFProducer/src/PFAlgo.cc

```diff
@@ -2717,8 +3118,14 @@ void PFAlgo::processBlock(const reco::PFBlockRef& blockref,
   elementLoop(block, linkData, elements, active, blockref, inds, deadArea);
 
   // Reconstruct pfCandidate from HF (either EM-only, Had-only or both)
+  // For phase2, process also pfblocks containing HF clusters and linked tracks
   if (!(inds.hfEmIs.empty() && inds.hfHadIs.empty())) {
-    createCandidateHF(block, blockref, elements, inds);
+    createCandidatesHF(block, linkData, elements, active, blockref, inds);
+    if (inds.hcalIs.empty() && inds.ecalIs.empty())
+      return;
+    edm::LogWarning("PFAlgo::processBlock")
+        << "Block contains HF clusters, but also contains ECAL or HCAL clusters. Continue.";
+    edm::LogWarning("PFAlgo::processBlock") << block;
```

Comment:
How frequent and unlikely is this condition?
LogWarnings are supposed to show some significant issues in the execution.
It looks like the condition for this warning is a property of the algorithm configuration, which means that this is not a warning but a debug.

Please issue only one LogWarning here, simply ```Continue.\n" << block``` should work here